src/sentry/api/endpoints/organization_traces.py

```diff
@@ -0,0 +1,160 @@
+        # exceeds 10_000 (the snuba query limit),
+        if limit > 10_000:
+            raise ParseError(detail="Too many spans per trace.")
+
+        def data_fn(offset: int, limit: int):
+            with handle_query_errors():
+                builder = SpansIndexedQueryBuilder(
+                    Dataset.SpansIndexed,
+                    cast(ParamsType, params),
+                    snuba_params=snuba_params,
+                    query=serialized.get("query", ""),
+                    selected_columns=["trace", "timestamp"] + serialized["field"],
+                    # The orderby is intentionally `None` here as this query is much faster
+                    # if we let Clickhouse decide which order to return the results in.
+                    # This also means we cannot order by any columns or paginate.
+                    orderby=None,
+                    limit=limit,
+                    offset=0,
+                    limitby=("trace", serialized["maxSpansPerTrace"]),
+                    sample_rate=options.get("traces.sample-list.sample-rate"),
+                    config=QueryBuilderConfig(
+                        transform_alias_to_input_format=True,
+                    ),
+                )
+                span_results = builder.run_query(Referrer.API_TRACE_EXPLORER_SPANS_LIST.value)
+                span_results = builder.process_results(span_results)
+
+            if not span_results["data"]:
+                return []
+
+            spans_by_trace: Mapping[str, list[Mapping[str, Any]]] = defaultdict(list)
+            for row in span_results["data"]:
+                spans_by_trace[row["trace"]].append(row)
+
+            trace_spans_count = sorted(
+                [(trace, len(spans)) for trace, spans in spans_by_trace.items()],
+                key=lambda item: item[0],
+                reverse=True,
+            )[:per_page]
+
+            spans_by_trace = {trace: spans_by_trace[trace] for trace, _ in trace_spans_count}
+
+            min_timestamp = snuba_params.end
+            max_timestamp = snuba_params.start
+            assert min_timestamp is not None
+            assert max_timestamp is not None
+            for spans in spans_by_trace.values():
+                for span in spans:
+                    timestamp = datetime.fromisoformat(span["timestamp"])
+                    if timestamp < min_timestamp:
+                        min_timestamp = timestamp
+                    if timestamp > max_timestamp:
+                        max_timestamp = timestamp
+
+            # TODO: move to use `update_snuba_params_with_timestamp`
+            time_buffer = options.get("performance.traces.transaction_query_timebuffer_days")
+            buffer = timedelta(days=time_buffer)
+            params["start"] = min_timestamp - buffer
+            params["end"] = max_timestamp + buffer
+            snuba_params.start = min_timestamp - buffer
+            snuba_params.end = max_timestamp + buffer
```

Comment:
We should replace this once Will gets min/max of traces working since we could just look it up.

Reply:
yep, ive got the TODO there already